packages/components/src/DataList/components/DataListAction/DataListAction.tsx

```diff
@@ -25,7 +25,16 @@ export function InternalDataListAction<T extends DataListObject>({
   const color = destructive ? "critical" : "blue";
 
   return (
-    <button className={styles.action} onClick={() => onClick?.(item)}>
+    <button
+      className={styles.action}
+      onClick={() => {
+        if (item) {
+          onClick?.(item);
+        } else {
+          onClick?.();
+        }
```

Comment:
I'm not really happy with this, open to suggestions

Reply:
first, put it in a handleClick function


```ts
function handleClick() {
  if (item) {
    return () => onClick?.(item)
  }

  return onClick
}
```